src/value_conversion.hpp

```diff
@@ -28,12 +28,14 @@ static Napi::Value GetValue(const Napi::CallbackInfo &info, size_t offset) {
 template <class T>
 class PointerHolder : public Napi::ObjectWrap<PointerHolder<T>> {
 public:
-	static Napi::FunctionReference *Init(Napi::Env env, const char *name) {
+	static void Init(Napi::Env env, Napi::Object exports, const char *name) {
 		auto func = Napi::ObjectWrap<PointerHolder<T>>::DefineClass(env, name, {});
-		auto constructor = new Napi::FunctionReference();
-		*constructor = Napi::Persistent(func);                     // weird
-		env.SetInstanceData<Napi::FunctionReference>(constructor); // is this so this is eventually freed?
-		return constructor;
+		constructor = Napi::Persistent(func); // set initial reference count to 1
+		exports.Set(name, func);
+		// How does the following interact with the static storage of constructor?
+		// Does not doing this create problems for multiple instance of this add-on, in multiple worker threads?
+		// See: https://github.com/nodejs/node-addon-api/blob/main/doc/object_wrap.md#example
+		// env.SetInstanceData<Napi::FunctionReference>(&constructor); // is this so this is eventually freed?
```

Comment:
It seems that storing this function reference in a static might be a problem is multiple instance of this add-on are created in the same Node process, in different environments.

But we can't store these constructors directly using SetInstanceData, because there is only one "data" slot per add-on instance, and we have multiple PointerHolders.

Ideas for how to do this better:
- Store a map from name to constructor in the instance data. This means we'd need to keep the name around in the PointerHolder (so we can look up the right constructor), but putting the name in a static seems fine. This is a bit more work to set up and a bit more overhead at runtime, but these are likely insignificant.
- Alternatively, since we're already storing the constructor in the exports, it seems like it should be possible to look it up using the environment (e.g. using Globals). I haven't quite figured that out, but I probably just haven't learned the right way.

Reply:
Oh, I just learned (by reading more [docs](https://github.com/nodejs/node-addon-api/blob/main/doc/addon.md)) that the Napi::Addon uses Napi::Env::SetInstanceData internally, so it shouldn't be used to store other data:

> Note: Napi::Addon<T> uses Napi::Env::SetInstanceData() internally. This means that the add-on should only use Napi::Env::GetInstanceData explicitly to retrieve the instance of the Napi::Addon<T> class. Variables whose scope would otherwise be global should be stored as instance variables in the Napi::Addon<T> class.

The recommendation is to store stuff directly on your Addon instance object, and use GetInstanceData to get the Addon instance:

```
void ExampleBinding(const Napi::CallbackInfo& info) {
  ExampleAddon* addon = info.Env().GetInstanceData<ExampleAddon>();
}
```

Reply:
Here's where node-addon-api puts the addon instance into the instance data: https://github.com/nodejs/node-addon-api/blob/cfcd2cf61d8d939bc2a2744af2ee72f07e4119c1/napi-inl.h#L6518